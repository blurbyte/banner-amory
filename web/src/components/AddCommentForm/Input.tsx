import styled from 'styled-components';

import BaseInput from '../Input';

const Input = styled(BaseInput).attrs({
  type: 'text',
  autoCapitalize: 'off',
  spellCheck: false,
  autoCorrect: 'off',
  autoComplete: 'off'
})`
  padding-left: 0;
`;

export default Input;
