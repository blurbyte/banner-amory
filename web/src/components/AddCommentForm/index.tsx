import * as React from 'react';
import { MutationFn } from 'react-apollo';

import { Comment, AddCommentInput } from '@sharedTypes/Comment';
import Form from './Form';
import Input from './Input';
import Actions from './Actions';
import DiscardButton from './DiscardButton';
import SubmitButton from './SubmitButton';

type AddCommentFormProps = {
  itemId: number;
  addComment: MutationFn<Comment, { input: AddCommentInput }>;
};

const initialState = {
  userName: '',
  message: ''
};

type AddCommentFormState = Readonly<typeof initialState>;

class AddCommentForm extends React.Component<AddCommentFormProps, AddCommentFormState> {
  readonly state = initialState;

  render() {
    return (
      <Form onSubmit={this.handleSubmit}>
        <Input
          placeholder="Name"
          name="userName"
          value={this.state.userName}
          onChange={this.handleInputChange}
        />
        <Input
          placeholder="Comment about this item..."
          name="message"
          value={this.state.message}
          onChange={this.handleInputChange}
        />
        {showActions(this.state) && (
          <Actions>
            <DiscardButton onClick={this.clearInputs} />
            <SubmitButton />
          </Actions>
        )}
      </Form>
    );
  }

  private handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    // trigger mutation
    this.props.addComment({
      variables: {
        input: {
          itemId: this.props.itemId,
          userName: this.state.userName,
          message: this.state.message
        }
      }
    });

    this.clearInputs();
  };

  private handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    } as AddCommentFormState);
  };

  private clearInputs = () => {
    this.setState(initialState);
  };
}

function showActions(state: AddCommentFormState) {
  return state.userName !== '' && state.message !== '';
}

export default AddCommentForm;
