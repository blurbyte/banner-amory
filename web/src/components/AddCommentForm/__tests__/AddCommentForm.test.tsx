import * as React from 'react';
import * as renderer from 'react-test-renderer';

import AddCommentForm from '../index';

test('renders correctly', () => {
  const component = renderer.create(<AddCommentForm itemId={83} addComment={jest.fn()} />);
  expect(component.toJSON()).toMatchSnapshot();
});
