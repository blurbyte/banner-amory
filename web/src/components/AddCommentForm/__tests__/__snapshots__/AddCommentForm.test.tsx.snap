// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="sc-bdVaJa fyQqOy"
  onSubmit={[Function]}
>
  <input
    autoCapitalize="off"
    autoComplete="off"
    autoCorrect="off"
    className="sc-htpNat epXOEy sc-bwzfXH bPLioj"
    name="userName"
    onChange={[Function]}
    placeholder="Name"
    spellCheck={false}
    type="text"
    value=""
  />
  <input
    autoCapitalize="off"
    autoComplete="off"
    autoCorrect="off"
    className="sc-htpNat epXOEy sc-bwzfXH bPLioj"
    name="message"
    onChange={[Function]}
    placeholder="Comment about this item..."
    spellCheck={false}
    type="text"
    value=""
  />
</form>
`;
